lib/travis/hub/model/log.rb

```diff
@@ -14,4 +14,12 @@ def clear
     update_column(:removed_by, nil)
     Part.where(log_id: id).delete_all
   end
+
+  def add(msg)
+    if aggregated_at
+      update_column(:content, [content, msg].join("\n"))
+    else
+      Part.create!(log_id: id, content: msg, final: true)
```

Comment:
Sending the log via HTTP probably makes sense, since that'll clear the contents already in the log, if any.